File: exercism/java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

public final class Gigasecond {
    private final LocalDateTime moment;

    public Gigasecond(final LocalDateTime moment) {
        this.moment = moment.plus(1_000_000_000, ChronoUnit.SECONDS);
    }

    public Gigasecond(final LocalDate moment) {
        this(moment.atStartOfDay());
    }

    public LocalDateTime getDateTime() {
        return moment;
    }
}

File: exercism/kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.temporal.ChronoUnit

class Gigasecond(date: LocalDateTime) {
    val date: LocalDateTime = date.plus(1_000_000_000, ChronoUnit.SECONDS)

    constructor(date: LocalDate) : this(date.atStartOfDay())
}
